dpnp/backend/kernels/dpnp_krnl_mathematical.cpp

```diff
@@ -281,37 +281,62 @@ template <typename _KernelNameSpecialization1, typename _KernelNameSpecializatio
 class dpnp_trapz_c_kernel;
 
 template <typename _DataType_input1, typename _DataType_input2, typename _DataType_output>
-void dpnp_trapz_c(void* array1_in, void* array2_in, void* result1, double dx, size_t array1_size, size_t array2_size)
+void dpnp_trapz_c(const void* array1_in, const void* array2_in, void* result1, double dx, size_t array1_size, size_t array2_size)
 {
-    _DataType_input1* array1 = reinterpret_cast<_DataType_input1*>(array1_in);
-    _DataType_input2* array2 = reinterpret_cast<_DataType_input2*>(array2_in);
+    if (!array1_size)
+    {
+        return;
+    }
+
+    cl::sycl::event event;
+    _DataType_input1* array1 = reinterpret_cast<_DataType_input1*>(const_cast<void*>(array1_in));
+    _DataType_input2* array2 = reinterpret_cast<_DataType_input2*>(const_cast<void*>(array2_in));
     _DataType_output* result = reinterpret_cast<_DataType_output*>(result1);
 
-    if (array2_size < 2) {
-        _DataType_input1* sum = reinterpret_cast<_DataType_input1*>(dpnp_memory_alloc_c(1 * sizeof(_DataType_input1)));
+    if (array1_size < 2) {
+        result[0] = 0;
 
-        dpnp_sum_c<_DataType_input1>(array1, sum, array1_size);
+        return;
+    }
 
-        result[0] = sum[0];
+    if (array1_size != array2_size) {
```

Comment:
You don't use `array2` here at all. How do you verify this algo? Are there tests for checking this?

Reply:
We need to use either `array2` or `dx`. `array2` is used in else-block, `dx` in if-block.

Reply:
In this case it is better update condition and restructured the code
```cpp
if (array1_size == array2_size)
{
    // use array2_size
}
else
{
    // use dx
}
```
This is more readable.